### PR TITLE
Feature/history detail

### PR DESCRIPTION
**Pull request by mkgzqd**

### CODE REVIEW

File: lib/pages/historyDetail.dart [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+              style: TextStyle(color: Colors.white),
+            ),
+          )
+        ]
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Set<Marker> markers = Set();
```
Comment (ryanfilkins97) [inline lib/pages/historyDetail.dart]: This markers set needs to be a class level variable if you want the markers to show up

File: lib/pages/historyDetail.dart [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                    zoom: 16.0
+                ),
+                markers: markers,
+                myLocationButtonEnabled: false,
+//              scrollGesturesEnabled: false,
+//              zoomGesturesEnabled: false,
+                rotateGesturesEnabled: false,
+              ),
+            ),
+            SizedBox(height: 20),
```
Comment (ryanfilkins97) [inline lib/pages/historyDetail.dart]: If you want the map to resize itself properly, you need to wrap everything below it in some sort of widget and give that widget the _bottomSheetKey global key. I recommend taking a look at the garage detail page for an example of this

File: lib/pages/historyDetail.dart [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                    fontSize: 20
+                ),
+              )
+            ],
+          ),
+          SizedBox(height: 20),
+          MaterialButton(
+            onPressed: () {
+              print("Reported");
+              Navigator.pushNamed(context, 'history/history_details/report_park', arguments: _park);
```
Comment (ryanfilkins97) [inline lib/pages/historyDetail.dart:144]: When testing this functionality, I got a crash related to the calendar widget:

RangeError (index): Invalid value: Only valid value is 0: 1

Not quite sure why this is

File: lib/pages/reportPark.dart [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+              color: getAppTheme().accentColor,
+              child: Text(
+                "REPORT",
+                style: TextStyle(color: Colors.white),
+              ),
+            )
+          ],
+        ),
+      ),
+      drawer: Drawer(
```
Comment (ryanfilkins97) [inline lib/pages/reportPark.dart:148]: The navbar should not be accessible from this page

File: lib/pages/reportPark.dart [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+      );
+
+      _scaffoldKey.currentState.showSnackBar(snackBar);
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedDate = _reportRepo.intervalStart;
```
Comment (ryanfilkins97) [inline lib/pages/reportPark.dart:58]: The default date should probably be the start date of the park passed in

File: lib/repositories/garageRepository.dart [hunk 6]
```diff
@@ -24,54 +24,54 @@ class GarageRepository {
 
   Future<List<Garage>> getGarages() {
     return Future(
-      () async {
```
Comment (ryanfilkins97) [inline lib/repositories/garageRepository.dart:27]: This file should not have been modified

**Review by ryanfilkins97 — CHANGES_REQUESTED** [review]
Requested changes in comments

File: lib/pages/historyDetail.dart (hunk 1 above)
Comment (mkgzqd) [reply lib/pages/historyDetail.dart]: Markers are now added.

File: lib/pages/historyDetail.dart (hunk 3 above)
Comment (mkgzqd) [reply lib/pages/historyDetail.dart:144]: Not sure where you're getting this crash. I can't replicate it.

File: lib/pages/reportPark.dart (hunk 4 above)
Comment (mkgzqd) [reply lib/pages/reportPark.dart:148]: I think the navbar should be accessible here for people to be able to navigate away from the page at any time. The Get Park function with the API call is called in the history page. In order for the Ticket Time to update on garage_detail, they must go through history so the API call is made and fetches the new data with the added  ticket time.

File: lib/pages/reportPark.dart (hunk 5 above)
Comment (mkgzqd) [reply lib/pages/reportPark.dart:58]: Fixed.

File: lib/pages/historyDetail.dart (hunk 3 above)
Comment (mkgzqd) [reply lib/pages/historyDetail.dart:144]: Are you working on ios? Some people here seem to encounter the same error and fixed it by running flutter clean. https://github.com/flutter/flutter/issues/17831

File: lib/repositories/garageRepository.dart (hunk 6 above)
Comment (mkgzqd) [reply lib/repositories/garageRepository.dart:27]: This should be the exact same besides formatting. Once I fix the other issue I'll try pulling down this file individually from master.

File: lib/pages/historyDetail.dart (hunk 3 above)
Comment (ryanfilkins97) [reply lib/pages/historyDetail.dart:144]: This is fixed for me now, I suspect it was related to the issue with the selected day in the reportPark page

File: lib/pages/historyDetail.dart (hunk 2 above)
Comment (mkgzqd) [reply lib/pages/historyDetail.dart]: I just added a widget with the _bottomSheetKey. Not sure if it has changed anything though.

Comment (ryanfilkins97) [reply lib/pages/historyDetail.dart]: For some reason _bottomSheetKey.currentContext is returning null when it should be allowing access to bottom sheet object, I'm really not sure why. Doesn't seem to crash the app, so might be worth revisiting after the demo

**Review by ryanfilkins97 — APPROVED** [review]

**Pull request merged**